test/dynamo/test_utils.py

```diff
@@ -484,6 +490,76 @@ def test2(x):
         self.assertEqual(compilation_events[0].ir_count, second)
 
+    @dynamo_config.patch(
+        {
+            "log_compilation_metrics": True,
+        }
+    )
+    def test_num_params(self):
+        import torch.nn as nn
+        import torch.nn.functional as F
+
+        class ModelSimple(nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.conv1 = nn.Conv2d(1, 20, 5)
+
+            def forward(self, x):
+                return F.relu(self.conv1(x))
+
+        self.assertEqual([x.numel() for x in ModelSimple().parameters()], [500, 20])
+
+        compilation_events = []
+        with mock.patch("torch._dynamo.utils.log_compilation_event") as log_event:
+            m = ModelSimple()
+            torch.compile(m)(torch.randn(1, 10, 10))
+            compilation_events = [arg[0][0] for arg in log_event.call_args_list]
+        self.assertEqual(compilation_events[0].param_numel, 520)
+        self.assertEqual(compilation_events[0].param_bytes, 4 * 520)
+        self.assertEqual(compilation_events[0].param_count, 2)
+
+        class ModelWrapped(nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.m1 = ModelSimple()
+                self.m2 = ModelSimple()
+
+            def forward(self, x):
+                return self.m1(x) + self.m2(x)
+
+        compilation_events = []
+        with mock.patch("torch._dynamo.utils.log_compilation_event") as log_event:
+            m = ModelWrapped()
+            torch.compile(m)(torch.randn(1, 10, 10))
+            compilation_events = [arg[0][0] for arg in log_event.call_args_list]
+        self.assertEqual(compilation_events[0].param_numel, 1040)
+        self.assertEqual(compilation_events[0].param_bytes, 4 * 1040)
+        self.assertEqual(compilation_events[0].param_count, 4)
+
+        @torch.compile()
+        def func(m1, m2, x):
+            return m1(x) + m2(x)
+
+        # Test a tied module
+        m = nn.Linear(4, 4)
+        with mock.patch("torch._dynamo.utils.log_compilation_event") as log_event:
+            func(m, m, torch.randn(4, 4))
+            compilation_events = [arg[0][0] for arg in log_event.call_args_list]
+        self.assertEqual(compilation_events[0].param_numel, 20)
+        self.assertEqual(compilation_events[0].param_bytes, 80)
+        self.assertEqual(compilation_events[0].param_count, 2)
+
+        # Test tied weights
+        m1 = nn.Linear(4, 4)
+        m2 = nn.Linear(4, 4)
+        m1.weight = m2.weight
+        with mock.patch("torch._dynamo.utils.log_compilation_event") as log_event:
+            func(m1, m2, torch.randn(4, 4))
+            compilation_events = [arg[0][0] for arg in log_event.call_args_list]
+        self.assertEqual(compilation_events[0].param_numel, 40)
+        self.assertEqual(compilation_events[0].param_bytes, 160)
+        self.assertEqual(compilation_events[0].param_count, 4)
```

Comment:
The tied weight test is not correct.

```suggestion
        self.assertEqual(compilation_events[0].param_numel, 24)
        self.assertEqual(compilation_events[0].param_bytes, 96)
        self.assertEqual(compilation_events[0].param_count, 3)
```